client/src/components/Button.tsx

```diff
@@ -2,12 +2,17 @@ import './Button.css';
 
 type Prop = {
   text: string;
-  onClick?: () => void;
+  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
 };
 
 export function Button({ text, onClick }: Prop) {
   return (
-    <button onClick={onClick} className="button">
+    <button
+      onClick={
```

Comment:
I'm surprised this is necessary. It should have worked before. However, in my mind, the cleanest way to do this is:
```
type Prop = {
  onClick?: () => void;
}

<button onClick={() => onClick?.()}>
```

That really strange syntax on `onClick` is the way to using optional chaining to call a function. If `onClick` is undefined, it won't make the function call.

Also, I've never had the need to examine the event when handling a button click, so using a void function should always be the right thing to do.